### ISSUE TITLE
Backport: Panic when attempting to view metrics for workload.

### ISSUE THREAD

**Issue by alena1108** [rancher/rancher#19371]
Backport https://github.com/rancher/rancher/issues/19358

**Comment by loganhz** [comment]
Steps to reproduce

1. Import a standalone  Pod by `Import YAML`.
2. Import a deployment.
3. View metrics of the deployment in Rancher UI

Results: 

You will see a red error popup

```
apiVersion: v1
kind: Pod
metadata:
  name: example
spec:
  containers:
    - name: busybox
      image: busybox:1.25
      volumeMounts:
        - name: quobytevolume
          mountPath: /persistent
      command:
        - /bin/sh
        - -ec
        - |
          if [ -f /persistent/log ]; then
            echo -n "Found old state starting at "; head -n1 /persistent/log
          else
            echo -n "Starting with a fresh state"
          fi
          while sleep 5; do date | tee -a /persistent/log; done
  volumes:
  - name: quobytevolume
    quobyte:
      registry: ignored:7861
      volume: testVolume
      readOnly: false
      user: root
      group: root
```

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  labels:
    app: nginx
spec:
  replicas: 3
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.7.9
        ports:
        - containerPort: 80

```

**Comment by alena1108** [comment]
Can be tested with rancher/rancher:v2.2.2-rc2

**Comment by jiaqiluo** [comment]
The bug is reproduced in Rancher v2.1.1
Setup: Rancher is single installed, add a DO cluster with 4 nodes(1 etcd, 1 control plane, 2 workers)
Following the steps in the previous comment from @loganhz , Rancher UI shows a red error message and the log shows a `panic serving api request` error.
<img width="338" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/55578558-127c5a00-56cb-11e9-96bc-fd7fc168107e.png">
```
2019/04/04 18:16:04 [ERROR] Panic serving api request:
goroutine 99923 [running]:
runtime/debug.Stack(0xc01834fb10, 0x33da1e0, 0x6a45ef0)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).ServeHTTP.func1(0x419c280, 0xc016e7fea0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:175 +0x67
panic(0x33da1e0, 0x6a45ef0)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/rancher/rancher/pkg/api/customization/monitor.parseMetricParams(0xc005224b00, 0x416bc80, 0xc014b65f58, 0xc00d359b08, 0x8, 0xc012f3c450, 0x7, 0xc00ed92c60, 0xc01962f280, 0x12, ...)
	/go/src/github.com/rancher/rancher/pkg/api/customization/monitor/handler_utils.go:253 +0x1594
github.com/rancher/rancher/pkg/api/customization/monitor.graph2Metrics(0xc005224b00, 0x4212840, 0xc0036fc000, 0xc012f3c450, 0x7, 0xc00d359b08, 0x8, 0xc01962f680, 0x1a, 0xc00b4051d0, ...)
	/go/src/github.com/rancher/rancher/pkg/api/customization/monitor/cluster_graph_action.go:146 +0xf3
github.com/rancher/rancher/pkg/api/customization/monitor.(*ProjectGraphHandler).QuerySeriesAction(0xc00121e9c0, 0xc019bac225, 0x5, 0xc0055aaf20, 0xc00e8585a0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/pkg/api/customization/monitor/project_graph_action.go:89 +0x79b
github.com/rancher/rancher/pkg/api/customization/monitor.(*ProjectGraphHandler).QuerySeriesAction-fm(0xc019bac225, 0x5, 0xc0055aaf20, 0xc00e8585a0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/pkg/api/server/managementstored/setup.go:511 +0x52
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.handleAction(0xc0055aaf20, 0xc00e8585a0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:263 +0x62
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).handle(0xc000aadd90, 0x419c280, 0xc016e7fea0, 0xc010781900, 0xc00a9ff200, 0xc005227ef0, 0xc0062e97d8)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:251 +0xd2
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).ServeHTTP(0xc000aadd90, 0x419c280, 0xc016e7fea0, 0xc010781900)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:180 +0x7d
github.com/rancher/rancher/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc004698a80, 0x419c280, 0xc016e7fea0, 0xc010781900)
	/go/src/github.com/rancher/rancher/vendor/github.com/gorilla/mux/mux.go:159 +0xf1
github.com/rancher/rancher/pkg/auth/requests.authHeaderHandler.ServeHTTP(0x41666c0, 0xc0057fdc20, 0x415bec0, 0xc004698a80, 0x417a340, 0xc003af5000, 0x4166680, 0xc005692630, 0x419c280, 0xc016e7fea0, ...)
	/go/src/github.com/rancher/rancher/pkg/auth/requests/filter.go:110 +0x60d
github.com/rancher/rancher/pkg/audit.auditHandler.ServeHTTP(0x415a800, 0xc005077640, 0x0, 0x419c280, 0xc016e7fea0, 0xc010781700)
	/go/src/github.com/rancher/rancher/pkg/audit/filter.go:31 +0x46a
github.com/rancher/rancher/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc004698a10, 0x419c280, 0xc016e7fea0, 0xc010781700)
	/go/src/github.com/rancher/rancher/vendor/github.com/gorilla/mux/mux.go:159 +0xf1
github.com/rancher/rancher/pkg/dynamiclistener.(*Server).cacheIPHandler.func1(0x419c280, 0xc016e7fea0, 0xc010781500)
	/go/src/github.com/rancher/rancher/pkg/dynamiclistener/server.go:419 +0x101
net/http.HandlerFunc.ServeHTTP(0xc0006a1260, 0x419c280, 0xc016e7fea0, 0xc010781500)
	/usr/local/go/src/net/http/server.go:1964 +0x44
net/http.serverHandler.ServeHTTP(0xc00769b450, 0x419c280, 0xc016e7fea0, 0xc010781500)
	/usr/local/go/src/net/http/server.go:2741 +0xab
net/http.(*conn).serve(0xc017399680, 0x41a5900, 0xc0148d0e00)
	/usr/local/go/src/net/http/server.go:1847 +0x646
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2851 +0x2f5
```

The bug fix is validated in Rancher v2.2.2-rc2
Setup: Rancher is single installed, add a DO cluster with 4 nodes(1 etcd, 1 control plane, 2 workers)
